### DISCUSSION TITLE
port from env variable

### DISCUSSION THREAD

**Discussion by skelawsky** [porsager/postgres#745] [Q&A]
How pass to config a variable from local enviroment? To be more precise a `port` variable?
```
import "dotenv/config";
export const migrationClient = postgres(
  "postgres://username:password@host:port/database",
  {
    host: process.env.DB_HOST,
    user: process.env.DB_USER,
    password: process.env.DB_PASSWORD,
    database: process.env.DB_NAME,
    port: Number(String(process.env.DB_PORT))
  }
);
```
Every other variable works, but as soon as I want pass `port` it stops saying `invalid url`

I store port in `.env` like this: `DB_PORT=5432`

**Comment by codeAdityagithub** [comment]
You don't need to explicitly add all the env options you can just have them in your .env in the following format:

PGHOST=127.0.0.1
PGPORT=5432
PGDATABASE=database
PGUSERNAME=postgres
PGPASSWORD=postgres

then user the following code for the config:
```
import postgres from "postgres";

const sql = postgres({}); // will use psql environment variables

export default sql;

```
